src/components/ChartContainer/index.js

```diff
@@ -47,12 +47,45 @@ class ChartContainer extends Component {
     return width - yAxisWidth * visibleAxesCount - margin.left - margin.right;
   };
 
+  getCalcSubDomain(series, hiddenSeries, xAxis, subDomain) {
+    let min = Number.MAX_SAFE_INTEGER;
+    let max = 0;
+    Object.keys(series)
+      .filter(key => !hiddenSeries[key])
+      .forEach(key => {
+        const serie = series[key];
+        const xAccessor = serie.xAccessor || xAxis.accessor;
+        const data = serie.data.filter(
```

Comment:
If we know that the data is ordered, we could be clever and stop looking once `xAccessor(d) > subDomain[1]`

Similarly, we should probably only do `xAccessor(d)` once

Reply:
Isn't is that code readability over small performance optimization? Consider the following code
```
const data = [];
for (let i = 0; i < serie.data.length; i += 1) {
  const d = serie.data[i];
  const value = xAccessor(d);

  if (subDomain[0] < value && value < subDomain[1]) {
    data.push(d);
  }

  if (value > subDomain[1]) {
    break;
  }
}
```
vs
```
const data = serie.data.filter(d => {
  const value = xAccessor(d);
  return subDomain[0] < value && value < subDomain[1];
});
```

Reply:
\*shrug\*

Considering we bill this as a high-performance charting library, we should consider performance optimizations, especially if they're happening on `render()`.

With that being said: I'm okay leaving this one and revisiting it once we're in the optimization phase.

Reply:
That's a good idea and thank you for pointing out.